source/editor.c

```diff
@@ -9,13 +9,58 @@ which these 4 will have to undergo in order to prove themselves worthy? Stay tun
 #include <stdio.h>
 #include <string.h>
 #include <stdlib.h>
+#define MAX 10
 
 struct node{
 	char data;
 	struct node *next;
 	struct node *prev;
 };
 
+struct URNODE{
+	struct node *link;
+	char operation;
+};
+
+int UFRONT=-1,UREAR=-1,RFRONT=-1,RREAR=-1;
+struct URNODE UNDO[10],REDO[10];
+
+//undo push
+
+void Upush(){
```

Comment:
Naming conventions. You should follow a certain convention for naming. Ideally do this: camelCase for functions, underscore_case for variables.